### PR TITLE
docs:Updated bpf documentation files

### PR DESCRIPTION
**Pull request by AdityaK60**
Grammatically updated the documentations of bpf.

Signed-off-by: Aditya Kumar <aditya.kumar60@infosys.com>

### CODE REVIEW

File: Documentation/bpf/architecture.rst [hunk 1]
```diff
@@ -33,7 +33,7 @@ in the networking stack, the stack has not yet extracted metadata out of the
 packet. On the other hand, tc BPF programs are executed later in the kernel
 stack, so they have access to more metadata and core kernel functionality.
 Apart from tc and XDP programs, there are various other kernel subsystems as
-well which use BPF such as tracing (kprobes, uprobes, tracepoints, etc).
+well, which use BPF such as tracing (kprobes, uprobes, tracepoints, etc).
```
Comment (qmonnet) [inline Documentation/bpf/architecture.rst]: We should as well fix the sentence here, “which use BPF such as tracing” is not very clear. And the “as well” in addition to “other” is a bit redundant. How about:

“Apart from tc and XDP programs, various other kernel subsystems use BPF, such as tracing (via kprobes, uprobes, tracepoints, for example).”

File: Documentation/bpf/architecture.rst [hunk 2]
```diff
@@ -81,7 +81,7 @@ The advantages for pushing these instructions into the kernel include:
   programs, for example, reuse the existing in-kernel drivers and operate on the
   provided DMA buffers containing the packet frames without exposing them or an entire
   driver to user space as in other models. Moreover, XDP programs reuse the existing
-  stack instead of bypassing it. BPF can be considered a generic "glue code" to
+  stack instead of bypassing it. BPF can be considered a generic "glue code" for
```
Comment (qmonnet) [inline Documentation/bpf/architecture.rst]: Double `for` makes it hard to parse, maybe?

```suggestion
  stack instead of bypassing it. BPF can be considered a generic "glue code" between
```

File: Documentation/bpf/architecture.rst [hunk 3]
```diff
@@ -148,7 +148,7 @@ which, by design, means that any program will terminate quickly. For kernel newe
 than 5.1 this limit was lifted to 1 million BPF instructions. Although the
 instruction set contains forward as well as backward jumps, the in-kernel BPF
 verifier will forbid loops so that termination is always guaranteed. Since BPF
-programs run inside the kernel, the verifier's job is to make sure that these are
+programs run inside the kernel; the verifier's job is to make sure that these are
```
Comment (qmonnet) [inline Documentation/bpf/architecture.rst]: No, I don't think that's correct. Why would we have a semicolon here?

File: Documentation/bpf/architecture.rst [hunk 4]
```diff
@@ -305,7 +305,7 @@ part of the core kernel and cannot be extended or added through kernel modules.
 
 The aforementioned function signature also allows the verifier to perform type
 checks. The above ``struct bpf_func_proto`` is used to hand all the necessary
-information which need to be known about the helper to the verifier, so that
+information which needs to be known about the helper to the verifier, so that
```
Comment (qmonnet) [inline Documentation/bpf/architecture.rst]:
```suggestion
information that needs to be known about the helper to the verifier, so that
```

File: Documentation/bpf/progtypes.rst [hunk 5]
```diff
@@ -263,7 +263,7 @@ cases.
 ..
 
 One example of XDP BPF production usage is Facebook's SHIV and Droplet
-infrastructure which implement their L4 load-balancing and DDoS countermeasures.
+infrastructure which implements their L4 load-balancing and DDoS countermeasures.
```
Comment (qmonnet) [inline Documentation/bpf/progtypes.rst:266]: I'd rather add an “s” to “infrastructure” instead, as [SHIV and Droplet](https://netdevconf.org/2.1/slides/apr6/zhou-netdev-xdp-2017.pdf) are two different things.

File: Documentation/bpf/toolchain.rst [hunk 6]
```diff
@@ -629,7 +629,7 @@ A full command line example with llc's ``-mcpu=probe``:
     $ llc xdp-example.bc -march=bpf -mcpu=probe -filetype=obj -o xdp-example.o
 
 Generally, LLVM IR generation is architecture independent. There are
-however a few differences when using ``clang --target=bpf`` versus
+however, a few differences when using ``clang --target=bpf`` versus
```
Comment (qmonnet) [inline Documentation/bpf/toolchain.rst:632]: “There are however,” sounds weird to me if you only have a comma _after_ “however”. Wouldn't you need one before as well? “There are, however,” or even “However/Nonetheless, there are a few differences...”.

File: Documentation/bpf/toolchain.rst [hunk 7]
```diff
@@ -682,7 +682,7 @@ following sample code:
           *c = sum;
         }
 
-At default code generation, the assembler will looks like:
+At default code generation, the assembler will look like:
```
Comment (qmonnet) [inline Documentation/bpf/toolchain.rst]: Since we're fixing this, let's just use present tense, “looks like”.

File: Documentation/bpf/toolchain.rst [hunk 8]
```diff
@@ -698,7 +698,7 @@ At default code generation, the assembler will looks like:
 
 64-bit registers are used, hence the addition means 64-bit addition. Now, if you
 enable the new 32-bit subregisters support by specifying ``-mattr=+alu32``, then
-the assembler will looks like:
+the assembler will look like:
```
Comment (qmonnet) [inline Documentation/bpf/toolchain.rst]: “looks like”

File: Documentation/bpf/toolchain.rst [hunk 9]
```diff
@@ -1775,7 +1775,7 @@ of all details, but enough for getting started.
   It basically means that the ``bpf`` classifier does not need to call into external
   tc action modules, which are not necessary for ``bpf`` anyway, since all packet
   mangling, forwarding or other kind of actions can already be performed inside
-  the single BPF program which is to be attached, and is therefore significantly
+  the single BPF program, which is to be attached, and is therefore significantly
```
Comment (qmonnet) [inline Documentation/bpf/toolchain.rst]: No, adding the comma here reads as if “the single BPF program” is “significantly faster”, when the real subject is “all packet mangling, forwarding or other kind of actions”.

Let's remove “which is to be attached” entirely, it doesn't bring any useful info here:

```suggestion
  a single BPF program, and is therefore significantly
```

File: Documentation/bpf/toolchain.rst [hunk 10]
```diff
@@ -1864,7 +1864,7 @@ of all details, but enough for getting started.
     # tc qdisc del dev em1 clsact
 
   tc BPF programs can also be offloaded if the NIC and driver has support for it
-  similarly as with XDP BPF programs. Netronome's nfp supported NICs offer both
+  similarly, as with XDP BPF programs. Netronome's nfp supported NICs offer both
```
Comment (qmonnet) [inline Documentation/bpf/toolchain.rst]: Doesn't sound correct. I think this should be “... if the NIC and driver have support for it, like XDP BPF programs.”

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thanks! Some comments inline below.

**Comment by qmonnet** [comment]
And please, get rid of the merge commit

**Comment by AdityaK60** [comment]
Raising a new PR with all the corrections suggested.

**Comment by AdityaK60** [comment]
Raising a new PR with all the corrections suggested.
